### PR TITLE
Add librime and its depend

### PR DESCRIPTION
**Pull request by tumashu**

### CODE REVIEW

**Comment by tumashu** [comment]
Rime is a input method, librime is its core library, more detail: https://github.com/rime/librime

**Comment by tumashu** [comment]
liberime is a module of emacs, which provide librime bind to elisp

**Comment by tumashu** [comment]
The last rebase fix  some  (mingw-w64) and license warn.

**Comment by tumashu** [comment]
@lazka What should I do next？

File: mingw-w64-leveldb/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+check() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+}
+
+package() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+  make DESTDIR="${pkgdir}" install
+  INSTALL -Dm644 ${srcdir}/${_realname}-${pkgver}/LICENSE ${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE
```
Comment (Alexpux) [inline mingw-w64-leveldb/PKGBUILD:50]: Install cammand must be lowercase everywhere

Comment (tumashu) [reply mingw-w64-leveldb/PKGBUILD]: Fixed.

File: mingw-w64-liberime/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  make
+}
+
+check() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+}
+
+package() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+  install -Dm644 liberime-core.dll ${pkgdir}${MINGW_PREFIX}/bin/liberime-core.dll
```
Comment (Alexpux) [inline mingw-w64-liberime/PKGBUILD:50]: cmake doesnt generate install targets?

File: mingw-w64-librime-data/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+source=("plum-git::git+https://github.com/rime/plum.git")
+sha256sums=('SKIP')
+
+build() {
+  cd ${srcdir}
+    
+  [[ -d build-${MINGW_CHOST} ]] && rm -rf build-${MINGW_CHOST}
+  mkdir build-${MINGW_CHOST}
+  
+  export rime_dir="${srcdir}/build-${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-librime-data/PKGBUILD:22]: if there are no build step then just drop build procedure and install directly in pkgdir in package procedure

File: mingw-w64-liberime/PKGBUILD (hunk 2 above)
Comment (tumashu) [reply mingw-w64-liberime/PKGBUILD:50]: yes

File: mingw-w64-librime/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+check() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+}
+
+package() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
+  make DESTDIR="${pkgdir}" install
+  install -Dm644 ${srcdir}/build-${MINGW_CHOST}/lib/librime.dll ${pkgdir}${MINGW_PREFIX}/bin/librime.dll
```
[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime/PKGBUILD:62]: no, copy librime.dll to bin path，is let liberime can find it, it is workaround for the liberime.

[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime/PKGBUILD:62]: liberime properly install librime.dll to lib path

File: mingw-w64-librime-data/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=0.0.1
+pkgrel=1
+pkgdesc="Rime input schema from plum (mingw-w64)"
+url="https://github.com/rime/plum"
+arch=('any')
+license=('LGPL3')
+makedepends=("git")
+source=("plum-git::git+https://github.com/rime/plum.git")
```
[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD:13]: plum.git is a download tool repo, we clone it then run rime-install script to download rime schema from other schema-repo

[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD:13]: maybe we should pkg every schema-repo, but it need much more work, while rime schema is just yaml data file.

File: mingw-w64-librime-data/PKGBUILD (hunk 3 above)
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD:22]: does running a download script mean build step？

File: mingw-w64-liberime/PKGBUILD (hunk 2 above)
Comment (tumashu) [reply mingw-w64-liberime/PKGBUILD:50]: at the moment, emacs module is new thing，emacs community seem to have no stand install path for module.

File: mingw-w64-librime-data/PKGBUILD (hunk 5 above)
[the remark this answers is not in the document]
Comment (Alexpux) [reply mingw-w64-librime-data/PKGBUILD:13]: I mean that you need somehow provide proper pkgver for package. If there are no release version for repo then you other aproaches

File: mingw-w64-librime/PKGBUILD (hunk 4 above)
[the remark this answers is not in the document]
Comment (Alexpux) [reply mingw-w64-librime/PKGBUILD:62]: For what reason dll installed into lib folder?

File: mingw-w64-librime-data/PKGBUILD (hunk 3 above)
Comment (Alexpux) [reply mingw-w64-librime-data/PKGBUILD:22]: downloading i think must be in prepare step

File: mingw-w64-librime/PKGBUILD (hunk 4 above)
[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime/PKGBUILD:62]: i see ...

[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime/PKGBUILD]: Fixed

File: mingw-w64-librime-data/PKGBUILD (hunk 5 above)
[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD]: I decide do not package this pkg at the moment, when i find new proper pkg way, i will pr

**Comment by tumashu** [comment]
ref archlinux's pkgname and pkgdesc

File: mingw-w64-liberime/PKGBUILD (hunk 2 above)
Comment (tumashu) [reply mingw-w64-liberime/PKGBUILD:50]: I have packaged a liberime-data-mini to, using the similar approach of archlinux. this problem should has been resolved.

File: mingw-w64-librime-data/PKGBUILD (hunk 3 above)
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD]: Fixed

File: mingw-w64-librime-data/PKGBUILD (hunk 5 above)
[the remark this answers is not in the document]
Comment (tumashu) [reply mingw-w64-librime-data/PKGBUILD]: I have packaged a librime-data-min, using archlinux's similar approach, this problem should fixed.

**Comment by tumashu** [comment]
@Alexpux please review again, thanks very much!

**Comment by tumashu** [comment]
rime-prelude create two package:  rime-prelude and librime-data-mini

**Comment by tumashu** [comment]
I have removed librime-data-mini and add librime-data package, similar archlinux's librime-data

I hope this is the latest rebase :-),  please review again, thanks! @Alexpux

**Comment by tumashu** [comment]
@Alexpux  Thanks very much,  How long will it take to enter the msys2 repo? https://packages.msys2.org/new

**Pull request merged**